src/containers/navigation-icons/AccountIcon.tsx

```diff
@@ -0,0 +1,60 @@
+import { useCallback, FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useAppSelector } from '~/hooks/use-redux'
+import { AxiosResponse } from 'axios'
+
+import Avatar from '@mui/material/Avatar'
+import Tooltip from '@mui/material/Tooltip'
+
+import { userService } from '~/services/user-service'
+import useAxios from '~/hooks/use-axios'
+import { defaultResponses } from '~/constants'
+
+import { styles } from '~/containers/navigation-icons/NavigationIcons.styles'
+
+import { UserResponse, UserRole } from '~/types'
+
+interface AccountIconProps {
+  openMenu: () => void
+}
+
+const AccountIcon: FC<AccountIconProps> = ({ openMenu }) => {
+  const { t } = useTranslation()
+  const { userId, userRole } = useAppSelector((state) => state.appMain)
+
+  const getUserData: () => Promise<AxiosResponse<UserResponse>> = useCallback(
+    () => userService.getUserById(userId, userRole as UserRole),
+    [userId, userRole]
+  )
+
+  const { loading, response } = useAxios<UserResponse>({
+    service: getUserData,
+    fetchOnMount: true,
+    defaultResponse: defaultResponses.object as UserResponse
+  })
+
+  const photoUrl =
+    response.photo &&
+    `${import.meta.env.VITE_APP_IMG_USER_URL}${response.photo}`
+
+  const userNameForAvatar = loading
+    ? ''
+    : `${response.firstName?.slice(0, 1) || ''}${
+        response.lastName?.slice(0, 1) || ''
+      }`
```

Comment:
I think using `if` check would be more cleaner solution here. I suggest refactoring like this:

```suggestion
let usernameInitials = "";
if (loading && response) {
    usernameInitials = `${response.firstName[0] ?? ""}${response.lastName[0] ?? ""}`
}
```

I am not sure about a need to use `?? ""` here. Please double check it.

Reply:
Why did you decide not to use `loading` variable to show spinner for example? Maybe it would be better to show some fallback when image and user data are loading

Reply:
Thank you! Realy using the nullish coalescing operator (??) generally considered better for handling default values in TypeScript

I've desided to use default Avatar when loading. Thanks!

![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/127191734/ddba7515-206d-48e8-a477-2d3648848ac9)